demos/formula_recognition_demo/python/utils.py

```diff
@@ -242,40 +243,50 @@ def _process_decoding_results(self):
             targets = np.argmax(logits, axis=1)
             self.model_status = Model.Status.READY
             return logits, targets
-        self._infer_request_handle_decoder = self._async_infer_decoder(self.row_enc_out,
-                                                                       self.dec_states_c,
-                                                                       self.dec_states_h,
-                                                                       self.output,
-                                                                       self.tgt,
-                                                                       req_id=0
-                                                                       )
+        self._async_infer_decoder(self.row_enc_out,
+                                  self.dec_states_c,
+                                  self.dec_states_h,
+                                  self.output,
+                                  self.tgt.squeeze(axis=0)
+                                  )
 
         return None
 
     def _start_encoder(self, model_input):
-        self._infer_request_handle_encoder = self._async_infer_encoder(model_input, req_id=0)
+        self._async_infer_encoder(model_input)
         self.model_status = Model.Status.ENCODER_INFER
 
     def _start_decoder(self):
-        enc_res = self._infer_request_handle_encoder.output_blobs
+        enc_res = self.infer_request_encoder.results
         self._unpack_enc_results(enc_res)
-        self._infer_request_handle_decoder = self._async_infer_decoder(
-            self.row_enc_out, self.dec_states_c, self.dec_states_h, self.output, self.tgt, req_id=0)
+        self._async_infer_decoder(self.row_enc_out, self.dec_states_c, self.dec_states_h, self.output, self.tgt)
         self.model_status = Model.Status.DECODER_INFER
 
     def _unpack_dec_results(self, dec_res):
-        self.dec_states_h = dec_res[self.args.dec_st_h_t_layer].buffer
-        self.dec_states_c = dec_res[self.args.dec_st_c_t_layer].buffer
-        self.output = dec_res[self.args.output_layer].buffer
-        logit = dec_res[self.args.logit_layer].buffer
+        for key, value in dec_res.items():
```

Comment:
possibly better to create mapping between layer_names and Node key in infer output once, instead of making choice after every inference